src/components/FilterPanel.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import { ConseilFilter } from '../Conseil';
+import '../style.css';
+
+interface FilterPanelProps {
+    filters: ConseilFilter;
+    setFilter:  (filters: ConseilFilter) => void;
+}
+
+export class FilterPanel extends React.Component<FilterPanelProps, ConseilFilter> {
+
+    constructor(props: FilterPanelProps) {
+        super(props);
+        this.state = props.filters;
+        this.handleBlockIDs = this.handleBlockIDs.bind(this);
+        this.handleLevels = this.handleLevels.bind(this);
+        this.handleNetIDs = this.handleNetIDs.bind(this);
+        this.handleProtocols = this.handleProtocols.bind(this);
+        this.handleOperationIDs = this.handleOperationIDs.bind(this);
+        this.handleAccountIDs = this.handleAccountIDs.bind(this);
+        this.handleAccountManagers = this.handleAccountManagers.bind(this);
+        this.handleAccountDelegates = this.handleAccountDelegates.bind(this);
+        this.handleLimit = this.handleLimit.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleBlockIDs(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'block_id': event.target.value.split(',')});
+    }
+
+    handleLevels(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'block_level': event.target.value.split(',').map(Number)});
+    }
+
+    handleNetIDs(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'block_netid': event.target.value.split(',')});
+    }
+
+    handleProtocols(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'block_protocol': event.target.value.split(',')});
+    }
+
+    handleOperationIDs(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'operation_id': event.target.value.split(',')});
+    }
+
+    handleOperationSources(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'operation_source': event.target.value.split(',')});
+    }
+
+    handleAccountIDs(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'account_id': event.target.value.split(',')});
+    }
+
+    handleAccountManagers(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'account_manager': event.target.value.split(',')});
+    }
+
+    handleAccountDelegates(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'account_delegate': event.target.value.split(',')});
+    }
+
+    handleLimit(event: React.ChangeEvent<HTMLInputElement>) {
+        this.setState({'limit': Number(event.target.value)});
```

Comment:
No need to put object keys in quotations in javascript 😄

Reply:
Ooh, changed!